tools/opensnoop.py

```diff
@@ -199,12 +256,70 @@
 }
 """
 
+bpf_text_kfunc_body_openat2 = """
+    u64 id = bpf_get_current_pid_tgid();
+    u32 pid = id >> 32; // PID is higher part
+    u32 tid = id;       // Cast and get the lower part
+    u32 uid = bpf_get_current_uid_gid();
+
+    PID_TID_FILTER
+    UID_FILTER
+    FLAGS_FILTER
+    if (container_should_be_filtered()) {
+        return 0;
+    }
+
+    struct data_t data = {};
+    bpf_get_current_comm(&data.comm, sizeof(data.comm));
+
+    u64 tsp = bpf_ktime_get_ns();
+
+    bpf_probe_read_user(&data.fname, sizeof(data.fname), (void *)filename);
+    data.id    = id;
+    data.ts    = tsp / 1000;
+    data.uid   = bpf_get_current_uid_gid();
+    data.flags = how->flags; // EXTENDED_STRUCT_MEMBER
+    data.ret   = ret;
+
+    events.perf_submit(ctx, &data, sizeof(data));
+
+    return 0;
+}
+"""
+
+b = BPF(text='')
+fnname_open = b.get_syscall_fnname("open").decode()
+fnname_openat = b.get_syscall_fnname("openat").decode()
+fnname_openat2 = b.get_syscall_fnname("openat2").decode()
```

Comment:
Like I mentioned above, may be test fnname_openat2? There is no need to test open and openat as openat exists since linux 2.6.16.

Reply:
Yes, indeed, I'll remove the checks against open and openat in next version.